AK/Optional.h

```diff
@@ -125,16 +125,35 @@ requires(!IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
         if (other.has_value())
             construct_at<RemoveConst<T>>(&m_storage, other.release_value());
     }
-    // Note: Checking for Move-Constructible to allow for reference members in T
+
+    // Note: The MoveConstructible only versions are to allow for non-move assignable types,
+    //       Such as types containing a reference, as move assigning them into an Optional
```

Comment:
nit: There is an uppercase after a comma, and the sentence is missing a final dot.
And you might want to split this long sentence into two.

While I'm here, I already liked this `constexpr` `Optional` but there is a side effect that I really appreciate too: gdb is now capable of correctly interpreting the inner type. Thanks for your work!

Reply:
Reworded it a bit, hope it's better now